src/super_gradients/training/utils/visualization/pose_estimation.py

```diff
@@ -10,55 +10,197 @@ def draw_skeleton(
+        keypoints = keypoints[..., 0:2].astype(int)
+        show_keypoint_confidence = False
 
-    for keypoint, color in zip(keypoints, keypoint_colors):
-        color = tuple(map(int, color))
-        cv2.circle(image, tuple(keypoint[:2]), radius=keypoint_radius, color=color, thickness=-1, lineType=cv2.LINE_AA)
+    keypoints_to_show_mask = keypoint_scores > keypoint_confidence_threshold
 
-    for joint, color in zip(edge_links, edge_colors):
-        p1 = tuple(keypoints[joint[0]][:2])
-        p2 = tuple(keypoints[joint[1]][:2])
+    pose_center = keypoints[keypoints_to_show_mask].mean(axis=0)
+    direction_from_center = keypoints - pose_center
+    direction_from_center /= np.linalg.norm(direction_from_center, axis=1, ord=2, keepdims=True) + 1e-9
+
+    overlay = image.copy()
+
+    for keypoint, score, direction, show, color in zip(keypoints, keypoint_scores, direction_from_center, keypoints_to_show_mask, keypoint_colors):
+        if not show:
+            continue
+        x, y = keypoint
+        x = int(x)
+        y = int(y)
         color = tuple(map(int, color))
-        cv2.line(image, p1, p2, color=color, thickness=joint_thickness, lineType=cv2.LINE_AA)
+        cv2.circle(overlay, center=(x, y), radius=keypoint_radius, color=color, thickness=-1, lineType=cv2.LINE_AA)
+
+        # Draw confidence score for each keypoint individually
+        if show_keypoint_confidence:
+            center_of_score = keypoint + direction * 16
+            text = f"{score:.2f}"
+            (w, h), baseline = cv2.getTextSize(text, cv2.FONT_HERSHEY_SIMPLEX, 0.5, 1)
+
+            cx, cy = center_of_score
+            if direction[0] < -0.5:
+                x = int(cx - w)
+            elif direction[0] > 0.5:
+                x = int(cx)
+            else:
+                x = int(cx - w // 2)
+
+            y = int(cy + h // 2)
+            cv2.putText(overlay, text, org=(x, y), fontFace=cv2.FONT_HERSHEY_SIMPLEX, fontScale=0.5, color=(250, 250, 250), thickness=1, lineType=cv2.LINE_AA)
+
+    if edge_links is not None:
+        for (kp1, kp2), color in zip(edge_links, edge_colors):
+            show = keypoints_to_show_mask[kp1] and keypoints_to_show_mask[kp2]
+            if not show:
+                continue
+            p1 = tuple(map(int, keypoints[kp1]))
+            p2 = tuple(map(int, keypoints[kp2]))
+            color = tuple(map(int, color))
+            cv2.line(overlay, p1, p2, color=color, thickness=joint_thickness, lineType=cv2.LINE_AA)
+
+    confident_keypoints = keypoints[keypoints_to_show_mask]
+
+    if show_confidence and len(confident_keypoints):
+        x, y, w, h = cv2.boundingRect(confident_keypoints)
+        overlay = draw_bbox(overlay, title=f"{score:.2f}", box_thickness=box_thickness, color=(255, 0, 255), x1=x, y1=y, x2=x + w, y2=y + h)
+
+    return cv2.addWeighted(overlay, 0.75, image, 0.25, 0)
+
+
+class PoseVisualization:
+    @classmethod
+    def draw_poses(
```

Comment:
Why not simply a function `draw_poses` ?
Not fundamentally against the class + class_method, but I'm not really sure it adds clarity

Reply:
Did it for the purpose to stay in line with `DetectionVisualization` class.